Playgrounds/Homework3.playground/Sources/AnalyticsService.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+// AnalyticsService is the protocol that defines the fundamental behaviour
+// for services related to analytics
+protocol AnalyticsService {
+    var datetime: Date { get }
+    mutating func logEvent<T:AnalyticEvent>(_ event: T)
+}
+
+// SomeAnalyticsService is an example of analytic service
+public struct SomeAnalyticsService: AnalyticsService {
+    let datetime = Date()
+    private var logs = [String]()
```

Comment:
can you create this as an array of AnalyticsEvent? Can you think about which type of array to use and what are the limits? (existential vs opaque)

Reply:
Sure thing. I thought the aim was to keep just the logs, tho.

If we want to hold the events, we should opt for an array w/ existential types (not sure if that's the right terminology) because we need to keep `any` type that conforms the `AnalyticalEvent`.